src/helpers/pivot/pivot_menu_items.ts

```diff
@@ -15,3 +15,30 @@ export const pivotProperties: ActionSpec = {
   },
   icon: "o-spreadsheet-Icon.PIVOT",
 };
+
+export const FIX_FORMULAS: ActionSpec = {
+  name: _t("Fix pivot formulas"),
+  execute(env) {
+    const position = env.model.getters.getActivePosition();
+    const cell = env.model.getters.getCorrespondingFormulaCell(position);
+    const pivotId = env.model.getters.getPivotIdFromPosition(position);
+    if (!cell || !pivotId) {
+      return;
+    }
+    const { sheetId, col, row } = env.model.getters.getCellPosition(cell.id);
+    const pivot = env.model.getters.getPivot(pivotId);
+    pivot.init();
+    env.model.dispatch("INSERT_PIVOT", {
+      sheetId,
+      col,
+      row,
+      pivotId,
+      table: pivot.getTableStructure().export(),
```

Comment:
doesn't it explode if the pivot isn't loaded ?

AFAICT, `isSpillPivotFormula` returns `true` on the main cell even if it's not loaded yet

Reply:
If the pivot is not loaded, it will explode yes. But in this action `pivot.init()` is called, so we're safe.
But I forgot Odoo, I will add a check that the pivot is loaded, it will make the job